sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/AbstractLNAdapter.java

```diff
@@ -492,36 +460,37 @@ public List<ResumedDataTemplate> getDAI(ResumedDataTemplate rDtt, boolean updata
             );
         List<ResumedDataTemplate> resumedDTTs = lNodeTypeAdapter.getResumedDTTs(rDtt);
 
-        // complete the list by the overridden information in DAI
-        resumedDTTs = resumedDTTs.stream()
-                .map(rDTT -> completeResumedDTTFromDAI(rDTT,updatable))
-                .filter(rDTT -> !updatable || (rDTT.isUpdatable()))
-                .collect(Collectors.toList());
-        return resumedDTTs;
+        resumedDTTs.forEach(this::overrideAttributesFromDAI);
+
+        if (updatableOnly){
+            return resumedDTTs.stream().filter(ResumedDataTemplate::isUpdatable).collect(Collectors.toList());
+        } else {
+            return resumedDTTs;
+        }
     }
 
-    protected ResumedDataTemplate completeResumedDTTFromDAI(ResumedDataTemplate rDtt, boolean updatable){
+    protected void overrideAttributesFromDAI(final ResumedDataTemplate rDtt){
         Optional<?> opDaiAdapter = findMatch(rDtt.getDoName(), rDtt.getDaName());
-        if(opDaiAdapter.isPresent()) {
-            AbstractDAIAdapter<?> daiAdapter = (AbstractDAIAdapter<?>) opDaiAdapter.get();
-            if(daiAdapter.isValImport() != null) {
-                rDtt.setValImport(daiAdapter.isValImport());
-            }
-            rDtt.setDaiValues(daiAdapter.getCurrentElem().getVal());
-            if (updatable && (daiAdapter.isValImport() == null || daiAdapter.isValImport())) {
-                boolean isSg = daiAdapter.getCurrentElem().getVal().stream()
-                        .anyMatch(tVal -> tVal.getSGroup() != null && tVal.getSGroup().intValue() > 0);
-
-                if (isSg) {
-                    IEDAdapter iedAdapter = parentAdapter.getParentAdapter();
-                    rDtt.setValImport(iedAdapter.isSettingConfig(parentAdapter.getInst())); // override
-                } else if (rDtt.getDaName().getFc() == TFCEnum.SG || rDtt.getDaName().getFc() == TFCEnum.SE) {
-                    log.warn("Inconsistencies in the SCD file (Setting group and DAI FC)!");
-                    rDtt.setValImport(false);
-                }
+        if (opDaiAdapter.isEmpty()){
+            return;
+        }
+        AbstractDAIAdapter<?> daiAdapter = (AbstractDAIAdapter<?>) opDaiAdapter.get();
```

Comment:
you can use JAVA 8 here if you want :

`protected void overrideAttributesFromDAI(final ResumedDataTemplate rDtt) {
        findMatch(rDtt.getDoName(), rDtt.getDaName())
                .map(iDataAdapter -> (AbstractDAIAdapter<LNAdapter>) iDataAdapter)
                .map(SclElementAdapter::getCurrentElem)
                .ifPresent(tdai -> {
                    rDtt.setDaiValues(tdai.getVal());
                    if (tdai.isValImport() != null) {
                        rDtt.setValImport(tdai.isValImport());
                    } else if (tdai.getVal().stream().anyMatch(tVal -> tVal.getSGroup() != null && tVal.getSGroup().intValue() > 0)) {
                        LDeviceAdapter lDeviceAdapter = this.parentAdapter;
                        IEDAdapter iedAdapter = lDeviceAdapter.getParentAdapter();
                        rDtt.setValImport(iedAdapter.isSettingConfig(lDeviceAdapter.getInst()));
                    } else if (rDtt.getDaName().getFc() == TFCEnum.SG || rDtt.getDaName().getFc() == TFCEnum.SE) {
                        log.warn("Inconsistencies in the SCD file (Setting group and DAI FC)!");
                        rDtt.setValImport(false);
                    }
                });
    }`